### DISCUSSION TITLE
Two ways to react to a click on a web button

### DISCUSSION THREAD

**Discussion by ChrisVonFrosta** [arendst/Tasmota#21234] [Q&A]
i now this [way ](https://tasmota.github.io/docs/Berry-Cookbook/#adding-a-button-to-the-main-menu)to reacting to a click on a web button.

But in the [ArtNet Example](https://github.com/arendst/Tasmota/blob/development/tasmota/berry/artnet/artnet_ui.be) is no 
`def websensor()`.
Instead it uses `def web_add_handler()` to add a POST route to ` page_artnet_ctl()`. There is no `def websensor() end`, but:
```
 def page_artnet_ctl()
    import webserver
    if !webserver.check_privileged_access() return nil end

    import persist
    import introspect
    
    try
      if webserver.has_arg("artnetapply")
```
what is the reason or benefit to do it this way?

**Comment by sfromis** [comment]
The explanation lies in where `page_artnet_ctl` is called, meaning that it is **not** simply a "web button".
